atc/db/resource.go

```diff
@@ -584,13 +587,29 @@ func (r *resource) PinVersion(rcvID int) (bool, error) {
 		return false, err
 	}
 
```

Comment:
Ah, seems like we never rolled back the transaction on error cases...

```suggestion
	 defer Rollback(tx)
```

Can you also add this to `Resource.UnpinVersion`, `Job.GetNextBuildInputs`, `VolumeRepository.RemoveMissingVolumes`, `Pipeline.LoadDebugVersionsDB`?

(that's quite a bit more than I was expecting to find...I wonder if we can have some automated check for ensuring we always `defer Rollback(tx)`...)

Reply:
Great catch! Thanks :D 

For the other ones do you think it would be worth making a separate PR for those?